File: go/solver/solver.go
package solver

// Solver is an AoC puzzle solver interface
type Solver interface {

	// ProcessInput of a puzzle.
	ProcessInput(content string) error

	// Part1 is the solution for the part 1 of the puzzle
	Part1() (string, error)

	// Part2 is the solution for the part 2 of the puzzle
	Part2() (string, error)
}

File: kotlin/src/main/kotlin/solver/Solver.kt
package solver

interface Solver{

    // processInput of a puzzle.
    fun processInput(content: String)

    // part1 is the solution for the part 1 of the puzzle
    fun part1(): String?

    // part2 is the solution for the part 2 of the puzzle
    fun part2(): String?

}